### ISSUE TITLE
Auto-rescue can be too aggressive in egg hunt mode

### ISSUE THREAD

**Issue by supertuxkart-sourceforge-migration** [supertuxkart/stk-code#605]
**Author: mbjornstk**

Humans know better when they want to be rescued, so leave angle/speed based rescue for AIs only. Getting autorescued can be frustrating, games are about scratching the paint without remorse.

With current code you get rescued when trying to do acrobatics (battle mode, discovering shortcuts, bumping around) or when you know that leaving the kart "auto stand up" is faster.

Migrated-From: https://sourceforge.net/apps/trac/supertuxkart/ticket/549 ( attachements can be found at http://supertuxkart.sourceforge.net/TracFilesBackup/549 )

**Comment by mglisse** [comment]
> The most important points are handled now (battle mode, overworld).

What about Easter egg hunts? You are *supposed* to go off-track in those.

**Comment by auriamg** [comment]
Auto-rescue should only happen in situations where you actually are in trouble, have you actually experienced any auto-rescues in appropriate situations? If so, where/when? I think auto-rescue is probably good enough by now that this ticket should be closed, I don't think we have nowhere near as many false positive rescues now as in earlier versions

**Comment by mglisse** [comment]
It is true that auto-rescue has improved, and I don't think I have had much trouble with it in race mode lately (though all the old 3rd party tracks are gone since the last update, so there may be a bias). But when you go exploring the map, which is what Easter egg hunt is all about, it is normal to explore slopes that would normally trigger a rescue. Before I found where the eggs were in minigolf, I had a number of situations where I was taking a while to reach an interesting place (say above the tunnel), only to have the game decide that I probably needed rescue :-( In beehive, even knowing where the eggs are, I still regularly get wrongly rescued. Sure, the kart may be at a weird angle, but it is getting back to a stable position alright, unless a rescue happens first.

I understand that Easter egg is low priority, and if you believe that disabling auto-rescue for this mode would cause more trouble than it would help, that's fine.

**Comment by auriamg** [comment]
Agreed we could perhaps disable it specifically in easter eggs mode

There still remains a question though ; even though we mention the rescue key in tutorials, a lot of people are unaware of its existence and entirely rely on auto-rescue. If we decide to do that we need to think about how to do it in a way that won't have people stuck. Maybe I'm worrying too much, though

**Comment by mglisse** [comment]
I agree that's the main drawback. One workaround I could think of (without going all the way to making it an option) was adding a "rescue" entry in the menu that appears when one presses ESC, which would first print "you know, there is a key for just this purpose (BACKSPACE by default, see options)" then rescue the kart. But that would clutter this important menu, be more work, might have issues with multiplayer mode (?), and some people might not think about ESC either... I guess an option would be easier, but I understand trying to minimize the number of options.

**Comment by qwertychouskie** [comment]
Or, in easter egg mode, when an auto-rescue would be triggered, instead put text on the screen saying "Press $RESCUEKEY to get rescued".
